src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -36,30 +38,34 @@ public AddCommand parse(String args) throws ParseException {
                 ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_NOTE,
                         PREFIX_TAG);
 
-        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_PHONE)
+        // (add)
+        if (args.isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_NO_PARAMETERS, AddCommand.MESSAGE_USAGE));
+        // (add John)
+        } else if (!argMultimap.containsAll(PREFIX_NAME, PREFIX_PHONE)
                 || !argMultimap.getPreamble().isEmpty()) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
-        }
-
-        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_NOTE, PREFIX_ADDRESS);
+            throw new ParseException(String.format(MESSAGE_INVALID_PARAMETER_FORMAT, AddCommand.MESSAGE_USAGE));
+        // (add p/99898888)
+        } else if (!argMultimap.contains(PREFIX_NAME)) {
+            throw new ParseException(String.format(MESSAGE_NAME_PARAMETER_MISSING, AddCommand.MESSAGE_USAGE));
+        // (add n/John)
+        } else if (!argMultimap.contains(PREFIX_PHONE)) {
+            throw new ParseException(String.format(MESSAGE_PHONE_PARAMETER_MISSING, AddCommand.MESSAGE_USAGE));
+        // (add n/John p/98988898...)
+        } else {
+            argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_NOTE,
+                    PREFIX_ADDRESS);
```

Comment:
Can this be refactored to avoid the nested `if` statements? So that the 'happy path' would not need to be nested inside an `else`.

e.g.
```
if (condition1) {
}

if (condition2) {
}
...

// If all checks pass, do happy path things
```